dynamic_rest/serializers.py

```diff
@@ -76,6 +80,49 @@ def data(self):
                 self._sideloaded_data = ReturnList(data, serializer=self)
         return self._sideloaded_data
 
+    def update(self, queryset, validated_data):
+        lookup_attr = getattr(self.child.Meta, 'update_lookup_field', 'id')
+
+        lookup_objects = {}
+        all_objects_update = {}
+        for entry in validated_data:
+            lookup_key = entry.pop(lookup_attr)
+            if lookup_key is not fields.empty:
+                lookup_objects[lookup_key] = entry
+            else:
+                all_objects_update.update(entry)
+
+        lookup_keys = lookup_objects.keys()
+
+        if not all((bool(_) and not inspect.isclass(_) for _ in lookup_keys)):
+            raise exceptions.ValidationError('Invalid lookup key value')
+
+        # Since this method is given a queryset which can have many
+        # model instances, first find all objects to update
+        # and only then update the models.
+        objects_to_update = queryset.filter(
+            **{'{}__in'.format(lookup_attr): lookup_keys}
+        )
+
+        if len(lookup_keys) != objects_to_update.count():
+            raise exceptions.ValidationError(
+                'Could not find all objects to update: {} != {}'
+                .format(len(lookup_keys), objects_to_update.count())
+            )
+
+        updated_objects = []
+        for object_to_update in objects_to_update:
+            lookup_key = getattr(object_to_update, lookup_attr)
+            data = lookup_objects.get(lookup_key)
+            # Use model serializer to actually update the model
+            # in case that method is overwritten.
+            updated_objects.append(self.child.update(object_to_update, data))
+
+        if all_objects_update:
+            queryset.update(**all_objects_update)
```

Comment:
One important caveat to doing a bulk update in this way is that `post_save` signals will not get fired -- its worth either calling that out explicitly in documentation, or calling `.save` in a loop instead.